### PR TITLE
feat: use the new api proxy and improve embed shortcode

### PR DESCRIPTION
**Pull request by hugomrdias**
This PR adds:

- Support for the new API proxy
  - The proxy will handle all the external resources, caching and transforming data to our needs
  - New repo with the Proxy will be published asap
- Code coverage and CI status is now handled by the proxy 
- Code blocks from github are now support in the `embed` shortcode using the Proxy
- Code blocks with `title` or `symbols` have permalinks

`{{<embed src="github:filecoin-project/specs-actors/actors/builtin/paych/paych_actor.go" lang="go" symbol="resolveAccount" title="Payment actor - Resolve Account">}}
`
Screenshot 2020-09-17 at 11 58 46

- Code blocks from github have links to the github source page
`{{<embed src="github:filecoin-project/specs-actors/actors/builtin/paych/paych_actor.go" lang="go" symbol="resolveAccount">}}`
Screenshot 2020-09-17 at 11 57 57

- All Hugo Modules removed except the theme

No more codecov timeouts or go modules errors in the CI 🥳. The Proxy server should always keep data cached but fresh.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,39 +8,40 @@ This is the [Filecoin Specification](https://github.com/filecoin-project/specs),
...
-    -   [`hint`](#hint)
--   [Math mode](#math-mode)
-    -   [Wrap `def`, `gdef`, etc.](#wrap-def-gdef-etc)
-    -   [Wrap inline math text with code blocks](#wrap-inline-math-text-with-code-blocks)
-    -   [Wrap math blocks with code fences](#wrap-math-blocks-with-code-fences)
--   [Front-matter](#front-matter)
--   [External modules](#external-modules)
--   [Solving Common problems](#solving-common-problems)
--   [References](#references)
+- [Filecoin Specification](#filecoin-specification)
```
Comment (olizilla) [inline README.md:11]: Some VS Code plugin insists on adding the h1 from this doc to the ToC... I had to turn it off locally. I don't think we want it in the ToC.

**Review by olizilla — APPROVED** [review]
This is great! It can go in as is. A couple of suggestions

1. UI nit: Can we have the UI for code examples match the look of figures, by adding a subtle border and reducing the font size a little.

Screenshot 2020-09-17 at 14 48 57

vs

Screenshot 2020-09-17 at 14 48 46

2: UX nit... If the src attribute of the embed shortcode was just the github url, authors could copy and paste from their browser url bar when adding new embeds, and the markdown would more explicilty show what it was refering to.

```
{{<embed src="github:filecoin-project/specs-actors/actors/builtin/cron/cron_actor.go"  lang="go">}}
# vs
{{<embed src="https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/cron/cron_actor.go"  lang="go">}}
```

it seems nice to have the source code be a little less tied to the details of how the shortcode is currently implemented.

**Comment by olizilla** [comment]
Oh nice, i see. Well yes, as you said, I think you could use the github URL and the api could give the permalink to the latest commit at build time that the code snippet was pulled from.

If an author wants to link to a specific commit that never changes, they can just use the URL for that commit on github.

**Comment by hugomrdias** [comment]
> the api could give the permalink to the latest commit at build time that the code snippet was pulled from.

Ah nice thats even better, didnt think about that. 💚 you are the best @olizilla

**Comment by hugomrdias** [comment]
btw 34s npm run build in the CI 🥳

**Pull request merged**